src/ckeditor.js

```diff
@@ -12,9 +14,56 @@
  * @singleton
  */
 
-CKEDITOR.define( function() {
+CKEDITOR.define( [ 'editor', 'mvc/collection', 'promise' ], function( Editor, Collection, Promise ) {
 	var CKEDITOR = {
 		/**
+		 * A collection containing all editor instances created.
+		 */
+		instances: new Collection(),
+
+		/**
+		 * Creates an editor instance for the provided DOM element.
+		 *
+		 * The creation of editor instances is an asynchronous operation, therefore a promise is returned by this
+		 * method.
+		 *
+		 *     CKEDITOR.create( '#content' );
+		 *
+		 *     CKEDITOR.create( '#content' ).then( function( editor ) {
+		 *         // Manipulate "editor" here.
+		 *     } );
+		 *
+		 * @param {String|Object} element An element selector or a DOM element, which will be the source for the
+		 * created instance.
+		 * @returns {Promise} A promise, which will be fulfilled with the created editor.
+		 */
+		create: function( element ) {
+			var that = this;
+
+			return new Promise( function( resolve, reject ) {
+				// If a query selector has been passed, transform it into a real element.
+				if ( typeof element == 'string' ) {
+					element = document.querySelector( element );
```

Comment:
BTW. This is an interesting story why it was implemented this way. AFAIR the spec authors had to make this move to make the chances for faster implementation higher. As you may know browsers match selectors from the end (from the last token). They first find all nodes matching the last token and then filter out the rest by using next tokens to the left. Selectors were previously used only for the stylesheets, so in the entire document, hence it made sense for performance reasons to implement it this way.

Therefore it was easier to browser vendors to implement `element.querySelector` in the way it is implemented, because it simply reused the existing algorithms. Otherwise, spec authors would risk that the new feature wouldn't be adopted and as you can imagine it's pretty critical.